WordPress/Classes/ViewRelated/Menus/MenusViewController.m

```diff
@@ -0,0 +1,945 @@
+        _menusService = service;
+    }
+    
+    return self;
+}
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.automaticallyAdjustsScrollViewInsets = NO;
+    self.navigationItem.title = NSLocalizedString(@"Menus", @"Title for screen that allows configuration of your site's menus");
+    self.view.backgroundColor = [WPStyleGuide greyLighten30];
+
+    self.scrollView.translatesAutoresizingMaskIntoConstraints = NO;
+    self.scrollView.backgroundColor = self.view.backgroundColor;
+    self.scrollView.keyboardDismissMode = UIScrollViewKeyboardDismissModeInteractive;
+    self.scrollView.alpha = 0.0;
+    
+    // add a bit of padding to the scrollable content
+    self.stackView.layoutMargins = UIEdgeInsetsMake(0, 0, 10, 0);
+    self.stackView.layoutMarginsRelativeArrangement = YES;
+    
+    self.headerViewController.delegate = self;
+    self.detailsViewController.delegate = self;
+    self.itemsViewController.delegate = self;
+    
+    [self setupSaveButtonItem];
+    [self setupLoadingView];
+    [self setupItemsLoadingLabel];
+    
+    [self syncWithBlogMenus];
+}
+
+- (void)setupSaveButtonItem
+{
+    UIBarButtonItem *button = [[UIBarButtonItem alloc] initWithTitle:NSLocalizedString(@"Save", @"Menus save button title") style:UIBarButtonSystemItemSave target:self action:@selector(saveBarButtonItemPressed:)];
+    self.navigationItem.rightBarButtonItem = button;
+    button.enabled = NO;
+    _saveButtonItem = button;
+}
+
+- (void)setupLoadingView
+{
+    WPNoResultsView *loadingView = [[WPNoResultsView alloc] init];
+    loadingView.titleText = NSLocalizedString(@"Loading Menus...", @"Menus label text displayed while menus are loading");;
+    _loadingView = loadingView;
+}
+
+- (void)setupItemsLoadingLabel
+{
+    UILabel *label = [[UILabel alloc] init];
+    label.font = [WPFontManager systemLightFontOfSize:14];
+    label.textColor = [WPStyleGuide darkBlue];
+    label.numberOfLines = 0;
+    [self.stackView addArrangedSubview:label];
+    [label.leadingAnchor constraintEqualToAnchor:self.stackView.leadingAnchor constant:MenusDesignDefaultContentSpacing].active = YES;
+    [label.trailingAnchor constraintEqualToAnchor:self.stackView.trailingAnchor constant:-MenusDesignDefaultContentSpacing].active = YES;
+    label.hidden = YES;
+    _itemsLoadingLabel = label;
+}
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender
+{
+    [super prepareForSegue:segue sender:sender];
+    
+    if ([segue.destinationViewController isKindOfClass:[MenuHeaderViewController class]]) {
+        _headerViewController = segue.destinationViewController;
+    } else if ([segue.destinationViewController isKindOfClass:[MenuItemsViewController class]]) {
+        _itemsViewController = segue.destinationViewController;
+    } else if ([segue.destinationViewController isKindOfClass:[MenuDetailsViewController class]]) {
+        _detailsViewController = segue.destinationViewController;
```

Comment:
For these, they're embedded within the storyboard `MenusViewController` scene. Being so, I figured it best to have these as `readonly` properties and set via the ivar.